scipy/signal/_signaltools.py

```diff
@@ -3416,8 +3416,12 @@ def resample_poly(x, up, down, axis=0, window=('kaiser', 5.0),
         max_rate = max(up, down)
         f_c = 1. / max_rate  # cutoff of FIR filter (rel. to Nyquist)
         half_len = 10 * max_rate  # reasonable cutoff for sinc-like function
-        h = firwin(2 * half_len + 1, f_c,
-                   window=window).astype(x.dtype)  # match dtype of x
+        if np.issubdtype(x.dtype, np.floating):
```

Comment:
Can `x` be complex?

Reply:
The testsuite doesn't appear to be sensitive to the change to complex handling--let me ping the folks in the ticket to see what they think the correct behavior for complex input should be: @magicgoose, @jeremycochoy, @pavelkomarov.

I might argue that an untested code path is already broken anyway, but that's perhaps a bit harsh vs. at least asking people what behavior makes sense.

Reply:
https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.resample_poly.html is not very clear about `complex` (_should_ it even be accepted, or throw an error?)

If it needs to be accepted, I may _guess_ that it can be reasonable to process real and imaginary components independently, as if it was just 2 component vector, or, say, left and right channel of a stereo sound track. But I'm not really a scientist. :smile: